src/Product/Contracts/ProductRepositoryContract.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace IndieHD\Velkart\Product\Contracts;
+
+use Illuminate\Database\Eloquent\Collection;
+use IndieHD\Velkart\Product\Product;
+
+interface ProductRepositoryContract
+{
+    public function list(string $order = 'id', string $sort = 'desc', array $columns = ['*']): Collection;
```

Comment:
In the interest of supporting more advanced sorting, it might be worth changing the `$order` and `$sort` arguments to arrays, too. Doing so would allow for sorting on more than one column, for example.

Reply:
Actually, it might make more sense for these two arguments to be combined into a single, associative array that defines both the columns and the order, e.g.:

```php
[
    'price' => 'asc',
    'foo' => 'desc',
    // ...
];
```

Then again, this weakens the contract and gives this method more responsibility than it should probably have.

Maybe we should just leave this alone for now, but think about how we can improve the repository filtering/sorting, in general, across all repositories, such that it's flexible and object-driven.